### DISCUSSION TITLE
LSP Error code 232: The Pipe is Being Closed

### DISCUSSION THREAD

**Discussion by partisani** [helix-editor/helix#9098] [Troubleshooting] [answered]
### Summary

For some reason, `rust-analyzer` seems to crash randomly when i am editing code, looking at the logs, i see this: `helix_lsp::transport [ERROR] rust-analyzer err: <- IO(Os { code: 232, kind: BrokenPipe, message: "The pipe is being closed." })`, Then, in the command line i see lots of Async jobs failing, it also freezes helix when i try to save a file.

### Reproduction Steps

*No reproduction steps, it just started to happen randomly*

### Helix log

Here are the logs: [helix.log](https://github.com/helix-editor/helix/files/13688065/helix.log)

(i included it in a file because the amount of lines is huge (400+), i just cleaned it and continued editing code till it died, and its already this big...)

### Platform

Win11

### Terminal Emulator

cmd.exe

### Installation Method

choco

### Helix Version

23.10

**Comment by gabydd** [comment, marked as answer]
This is probably a bug in rust-analyzer try updating your toolchain

**Reply by partisani** [reply]
Well, now it's fixed, there are some other problems but at least i can use helix now.

EDIT: No, i cannot.

**Comment by Lilith-In-Starlight** [comment]
I'm getting this issue in other LSPs as well, it is not rust-analyzer
